From b11dd571cbe57a3d93cce87db69d3dae98d4872e Mon Sep 17 00:00:00 2001
From: Simeon Marijon <simeon.marijon@foss.st.com>
Date: Tue, 20 Jun 2023 14:36:51 +0200
Subject: [PATCH 0433/1141] dt-bindings: nvmem: add binding for STM32MPX TAMP
 backup registers

Adds the binding documentation for TAMP backup registers.
The backup registers is a part of the TAMP device
of STM32 MPU.

Change-Id: Ib38a75dfc6a77d68df6b4c5b91dfafca13ee2bcc
Signed-off-by: Simeon Marijon <simeon.marijon@foss.st.com>
---
 .../bindings/nvmem/st,stm32-tamp-nvram.yaml   | 69 +++++++++++++++++++
 MAINTAINERS                                   |  5 ++
 2 files changed, 74 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/nvmem/st,stm32-tamp-nvram.yaml

diff --git a/Documentation/devicetree/bindings/nvmem/st,stm32-tamp-nvram.yaml b/Documentation/devicetree/bindings/nvmem/st,stm32-tamp-nvram.yaml
new file mode 100644
index 000000000000..8f4dc7649f74
--- /dev/null
+++ b/Documentation/devicetree/bindings/nvmem/st,stm32-tamp-nvram.yaml
@@ -0,0 +1,69 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/nvmem/st,stm32-tamp-nvram.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: STMicroelectronics STM32 backup registers as NVMEM
+
+description: |
+  This represents STM32 backup registers area presented as nvmem cells.
+  This contains various information (e.g. boot mode passed to bootloaders).
+
+maintainers:
+  - Siméon Marijon <simeon.marijon@foss.st.com>
+
+allOf:
+  - $ref: "nvmem.yaml#"
+
+properties:
+  compatible:
+    enum:
+      - st,stm32mp15-tamp-nvram
+      - st,stm32mp25-tamp-nvram
+
+  reg:
+    description:
+      TAMP Registers bank location and size.
+    maxItems: 1
+
+  '#address-cells':
+    const: 1
+
+  '#size-cells':
+    const: 1
+
+patternProperties:
+  "^tamp-bkp@[0-9a-f]+$":
+    type: object
+    required:
+      - reg
+
+required:
+  - compatible
+  - reg
+  - "#address-cells"
+  - "#size-cells"
+
+additionalProperties: false
+
+examples:
+  - |
+          nvram: nvram@5c00a100 {
+              #address-cells = <1>;
+              #size-cells = <1>;
+              reg = <0x5c00a100 0x80>;
+              compatible = "st,stm32mp15-tamp-nvram";
+
+              /* Data cells */
+              saes_secret_key: tamp-bkp@0 {
+                  /*see saes secret key feature */
+                  reg = <0x0 0x20>;
+              };
+
+              boot_mode: tamp-bkp@78 {
+                  /*see boot mode selection feature*/
+                  reg = <0x78 0x4>;
+              };
+          };
+...
diff --git a/MAINTAINERS b/MAINTAINERS
index 71391b1a4356..f8a73c9f5520 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -19735,6 +19735,11 @@ F:	Documentation/devicetree/bindings/iio/adc/st,stm32-dfsdm-adc.yaml
 F:	Documentation/devicetree/bindings/sound/st,stm32-*.yaml
 F:	sound/soc/stm/
 
+STM32 TAMP DRIVERS
+M:	Siméon Marijon <simeon.marijon@foss.st.com>
+S:	Maintained
+F:	Documentation/devicetree/bindings/nvmem/st,stm32-tamp-nvram.yaml
+
 STM32 TIMER/LPTIMER DRIVERS
 M:	Fabrice Gasnier <fabrice.gasnier@foss.st.com>
 S:	Maintained
-- 
2.39.2

